### PR TITLE
sse2: implements last remaining intrinsics

### PR DESCRIPTION
**Pull request by gwenn**
sse2 is complete.

### CODE REVIEW

**Comment by gwenn** [comment]
I don't know why `_mm_loadr_pd` is failing on Travis. It is not impacted by this PR !
I am going to fix `_mm_store_pd` on Windows.

**Comment by gnzlbg** [comment]
@gwenn this is failing on 32 bit builds: 

```shell
test x86::i586::sse2::tests::_mm_loadr_pd ... thread 'x86::i586::sse2::tests::_mm_loadr_pd' panicked at 'attempt to add with overflow', coresimd/src/x86/i586/sse2.rs:4055:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.
FAILED
```

Very similar code used to break the `i686-apple-darwin` (incrementing the `offset` in the `while` loops overflowed). IIRC the way I fixed it was by using `repr(align)` (see [here](https://github.com/gwenn/stdsimd/blob/ad12c27023e523144ec6910faf27dff203e5364d/coresimd/src/x86/i586/sse2.rs#L3881) ).

We should probably remove all similar looking code from the test suite.

**Comment by gwenn** [comment]
I hope I will understand when I will see your patch how I made `_mm_loadr_pd` failed without modifying it.

**Comment by gnzlbg** [comment]
@gwenn You did not break it, it was already broken, but the issue was not surfacing before. Either a new version of the compiler broke it (there was a change related to `simd` and `repr(transparent)` merged this week), or adding new tests/functions to the module did :/

**Comment by gwenn** [comment]
@gnzlbg Thanks for your explanation.

**Comment by gnzlbg** [comment]
@gwenn I've submitted a PR (#245) that _might_ fix this.

**Comment by gnzlbg** [comment]
@gwenn I've merged the test fixes, maybe the fix the issues here (I've restarted the build but it might be picking the old master commit so you might have to rebase).

**Comment by gnzlbg** [comment]
Now things look way better, the `i586`/`i686-unknown-linux-gnu` builds fail with:

```shell
x86::sse2::assert__mm_shuffle_pd_shufpd' panicked at 'too many instructions in the disassembly', stdsimd-test/src/lib.rs:322:8
```

You probably want to enable the `assert_instr` of `__mm_shuffle_pd` only for `x86_64` platforms but the tests should be green afterwards.

**Comment by gwenn** [comment]
Sorry,
I don't have any windows machine so I can't investigate.

File: coresimd/src/x86/i686/sse2.rs [hunk 1]
```diff
@@ -111,17 +184,53 @@ extern "C" {
...
     use stdsimd_test::simd_test;
 
+    use core::mem;
     use v128::*;
-    use v64::i32x2;
+    use v64::{__m64, i32x2, u32x2};
     use x86::i686::sse2;
 
+    #[simd_test = "sse2"]
+    unsafe fn _mm_add_si64() {
```
Comment (alexcrichton) [inline coresimd/src/x86/i686/sse2.rs]: It looks like this and the two tests below it are causing the failures on Windows, could a `#[cfg]` be added with a FIXME perhaps?

**Comment by gnzlbg** [comment]
This is all green :)

**Pull request merged**